From abf7e91a7fc779140e5be1875985caae94916c7e Mon Sep 17 00:00:00 2001
From: Ofer Heifetz <oferh@marvell.com>
Date: Thu, 11 Jan 2018 17:43:45 +0200
Subject: [PATCH 2175/2241] crypto: inside-secure: add eip197d TRC record word
 count

the eip197b and 197d differ in the FW expected transformation
record size in words, this commit add support for the eip197d

Change-Id: Ib640e4413a8719ca92bb5806bb545de904e417e0
Signed-off-by: Ofer Heifetz <oferh@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/48881
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Dmitri Epshtein <dima@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
---
 drivers/crypto/inside-secure/safexcel.c | 11 ++++++++---
 drivers/crypto/inside-secure/safexcel.h |  7 +++++--
 2 files changed, 13 insertions(+), 5 deletions(-)

diff --git a/drivers/crypto/inside-secure/safexcel.c b/drivers/crypto/inside-secure/safexcel.c
index 4b1b508..3e1cbf6 100644
--- a/drivers/crypto/inside-secure/safexcel.c
+++ b/drivers/crypto/inside-secure/safexcel.c
@@ -69,8 +69,13 @@ static int eip197_trc_cache_init(struct device *dev,
 	    rc_ht_factor, rc_ht_entry_cnt, rc_admn_ram_wc,
 	    rc_admn_ram_entry_cnt;
 
-	rc_rec1_wc = EIP197_CS_TRC_REC_WC;
-	rc_rec2_wc = EIP197_CS_TRC_LG_REC_WC;
+	if (priv->eip197_hw_ver == EIP197B) {
+		rc_rec1_wc = EIP197B_CS_TRC_REC_WC;
+		rc_rec2_wc = EIP197B_CS_TRC_LG_REC_WC;
+	} else {
+		rc_rec1_wc = EIP197D_CS_TRC_REC_WC;
+		rc_rec2_wc = EIP197D_CS_TRC_LG_REC_WC;
+	}
 
 	/* Determine the RC record size to use */
 	if (rc_rec2_wc > rc_rec1_wc)
@@ -234,7 +239,7 @@ static int eip197_trc_cache_init(struct device *dev,
 	       priv->base + EIP197_TRC_FREECHAIN);
 
 	/* Set Hash Table start */
-	reg = ((EIP197_CS_TRC_REC_WC << EIP197_TRC_RECORD_SIZE2_OFFSET)		|
+	reg = ((rc_rec1_wc << EIP197_TRC_RECORD_SIZE2_OFFSET)		|
 		(EIP197_RC_DMA_WR_COMB_DLY << EIP197_TRC_DMA_WR_COMB_DLY_OFFSET)|
 		(rc_record_cnt & EIP197_TRC_HASH_TABLE_START_MASK));
 	writel(reg, priv->base + EIP197_TRC_PARAMS2);
diff --git a/drivers/crypto/inside-secure/safexcel.h b/drivers/crypto/inside-secure/safexcel.h
index eeb4bce..d83af3f 100644
--- a/drivers/crypto/inside-secure/safexcel.h
+++ b/drivers/crypto/inside-secure/safexcel.h
@@ -52,8 +52,11 @@
 /* EIP-197 Classification Engine */
 /* configuration parameters */
 /* classification */
-#define EIP197_CS_TRC_REC_WC				59
-#define EIP197_CS_TRC_LG_REC_WC				73
+#define EIP197B_CS_TRC_REC_WC				59
+#define EIP197B_CS_TRC_LG_REC_WC			73
+#define EIP197D_CS_TRC_REC_WC				64
+#define EIP197D_CS_TRC_LG_REC_WC			80
+
 /* record cache */
 #define EIP197_RC_MAX_ENTRY_CNT				4096
 #define EIP197_RC_MIN_ENTRY_CNT				32
-- 
2.7.4

